### ISSUE TITLE
Few notes on ogv-decoder-video-callbacks.js

### ISSUE THREAD

**Issue by brianxautumn** [brion/ogv.js#454]
The y, u, and v buffers are all on a contiguous piece of memory. On the frame output you should be able to output it as 1 continuous array + the offsets/strides. Without emscripten, this is 3 unnecessary array copies.

Also the resolution can change on any frame. 
Lastly, you would only get 0 or 1 frames per each call to vpx_codec_decode.

**Comment by brion** [comment]
Yeah, will need to change this to support resolution switching. Could optimize the buffer copy to since we know the memory is contiguous, though it'll copy about the same number of bytes so I don't expect a huge perf difference. One less GC'd object though, that may help. :)

**Comment by brion** [comment]
Related issues:
* Resolution changing: https://github.com/brion/ogv.js/issues/354
